sdk/mobile-center/src/test/java/com/microsoft/azure/mobile/channel/DefaultChannelTest.java

```diff
@@ -1077,4 +1081,114 @@ public boolean matches(Object argument) {
             }
         }));
     }
+
+    @Test
+    public void removeGroupWhileCounting() throws Exception {
+
+        /* Set up mocking. */
+        final CountDownLatch latch = new CountDownLatch(1);
+        final Semaphore semaphore = new Semaphore(0);
+        Persistence mockPersistence = mock(Persistence.class);
+        DatabasePersistenceAsync mockPersistenceAsync = mock(DatabasePersistenceAsync.class);
+        whenNew(DatabasePersistenceAsync.class).withArguments(mockPersistence).thenReturn(mockPersistenceAsync);
+        doAnswer(new Answer<Object>() {
+
+            @Override
+            public Object answer(final InvocationOnMock invocation) throws Throwable {
+                new Thread() {
+
+                    @Override
+                    public void run() {
+
+                        /* Wait for add/remove calls to be done. */
+                        try {
+                            latch.await();
+                        } catch (InterruptedException e) {
+                            throw new RuntimeException(e);
```

Comment:
What if it throws a runtime exception, does the test fail?
It is running on another thread so I don't think it will fail.